### PR TITLE
[Intl] Update ICU to version 55

### PR DESCRIPTION
**Pull request by jakzal**
| Q | A |
| --- | --- |
| Branch | 2.3 |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #14260 |
| License | MIT |
| Doc PR | - |

The update script needed to be fixed to work with latest data files from ICU as well.

The only change I needed to do in stubs and tests is related to the way negative numbers are formatted for currencies (a change from `(£100)` to `-£100`).

Form tests needed an update as well, as some date formats have changed.

### CODE REVIEW

**Comment by jakzal** [comment]
Some form tests are failing.

status: needs work

**Comment by jakzal** [comment]
Form tests updated. Failure on appveyor is unrelated.

status: needs review

File: src/Symfony/Component/Intl/Resources/data/languages/sh.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-    "Version": "2.1.8.20",
+    "Version": "2.1.13.24",
     "Names": {
         "aa": "Afarski",
         "ab": "abhaski",
```
Comment (HeahDude) [inline src/Symfony/Component/Intl/Resources/data/languages/sh.json]: is the case here intended ?

Comment (jakzal) [inline src/Symfony/Component/Intl/Resources/data/languages/sh.json]: This is a generated file.

Comment (jakzal) [inline src/Symfony/Component/Intl/Resources/data/languages/sh.json]: It's better to review this PR commit by commit. "[Intl] Update the ICU data to version 55" is an update for generated files.

Comment (HeahDude) [inline src/Symfony/Component/Intl/Resources/data/languages/sh.json]: Ok thanks.

**Comment by HeahDude** [comment]
This is big. Great work!

**Comment by fabpot** [comment]
Thank you @jakzal.

**Comment by webmozart** [comment]
Awesome @jakzal, thanks so much! :)

**Pull request merged**